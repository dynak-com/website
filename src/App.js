import React, { Component } from 'react';
import Navbar from './components/navbar/navbar.component';
import { GlobalStyles, StyledAppContainer } from './app.styles';
import { Switch, Route } from 'react-router';
import MainPage from './pages/main-page/main-page.component';
import AboutPage from './pages/about-page/about-page.component';
import ProjectsPage from './pages/projects-page/projects-page.component';
import ContactPage from './pages/contact-page/contact-page.component';
import ErrorPage from './pages/error-page/error-page.component';
import Footer from './components/footer/footer.component';
import content from './content.json';

const contentData = JSON.parse(JSON.stringify(content));

class App extends Component {
  state = {
    lang: 'pl'
  };
  content = contentData.find(item => item.lang === this.state.lang);
  render() {
    return (
      <StyledAppContainer>
        <GlobalStyles />
        <Navbar content={this.content.nav} targetId='headerNavbarMenu' />
        <Switch>
          <Route
            exact
            path='/'
            component={() => <MainPage content={this.content.home} />}
          />
          <Route
            exact
            path='/about'
            component={() => <AboutPage content={this.content.about} />}
          />
          <Route
            exact
            path='/projects'
            component={() => <ProjectsPage content={this.content.projects} />}
          />
          <Route
            exact
            path='/contact'
            component={() => <ContactPage content={this.content.contact} />}
          />
          <Route component={() => <ErrorPage content={this.content.error} />} />
        </Switch>
        <Footer />
      </StyledAppContainer>
    );
  }
}

export default App;
