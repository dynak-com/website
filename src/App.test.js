import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import useComponent from './hook';

function HookWrapper(props) {
    const hook = props.hook ? props.hook() : undefined;
    return <HookWrapper hook={hook} />;
}

describe('App component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.exists()).toBeTruthy();
    });

    it('should set init value', () => {
        const wrapper = shallow(<HookWrapper hook={() => useComponent()} />);
        const hook = wrapper.props().hook;
        const { lang, onSetLang } = hook;
        expect(lang).toEqual('pl');
    });
});
