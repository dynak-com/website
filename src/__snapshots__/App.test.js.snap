// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly and contains the correct items 1`] = `
"<div id=\\"my\\" className=\\"main-container\\">
  <Navbar content={{...}} targetId=\\"headerNavbarMenu\\" />
  <Switch>
    <Route exact={true} path=\\"/\\" component={[Function: component]} />
    <Route exact={true} path=\\"/about\\" component={[Function: component]} />
    <Route exact={true} path=\\"/projects\\" component={[Function: component]} />
    <Route exact={true} path=\\"/contact\\" component={[Function: component]} />
    <Route component={[Function: component]} />
  </Switch>
  <Footer />
  <ModalsContainer modalsData={{...}} />
</div>"
`;
